js/components/tabs.js

```diff
@@ -110,20 +133,60 @@ var Tabs = React.createClass({
     var panel = this.props.children[index];
 
     return (
-      <article ref='tab-panel' className='tab-panel'>
+      <div ref='tab-panel' className='tab-panel' role={CONSTANTS.ARIA_ROLES.TAB_PANEL}>
         {panel}
-      </article>
+      </div>
     );
   },
 
   handleLeftChevronClick: function(event) {
     event.preventDefault();
-    this.refs.tabsNavigation.scrollLeft -= 30;
+    if (this.tabsNavigationElement) {
+      this.tabsNavigationElement.scrollLeft -= 30;
+    }
   },
 
   handleRightChevronClick: function(event) {
     event.preventDefault();
-    this.refs.tabsNavigation.scrollLeft += 30;
+    if (this.tabsNavigationElement) {
+      this.tabsNavigationElement.scrollLeft += 30;
+    }
+  },
+
+  onMenuItemFocus: function(event) {
+    if (event.currentTarget) {
+      this.scrollIntoViewIfNeeded(event.currentTarget);
+    }
+  },
+
+  /**
+   * Ensures that the given menu item is completely visible inside the tabs navigation.
+   * Adjusts the tabs navigation's scroll position when this is not the case.
+   * @private
+   * @param {HTMLElement} menuItem The menu item which we want to make sure is visible.
+   */
+  scrollIntoViewIfNeeded: function(menuItem) {
+    if (
+      !this.tabsNavigationElement ||
+      !menuItem ||
+      typeof menuItem.clientWidth === 'undefined'
+    ) {
+      return;
+    }
+    // Element is at a position that starts before the current navigation's scroll
+    // position. This will cause the element to be cut-off, so we set the scroll position
+    // to the value of the element's offset.
+    if (menuItem.offsetLeft < this.tabsNavigationElement.scrollLeft) {
+      this.tabsNavigationElement.scrollLeft = menuItem.offsetLeft;
+    } else {
+      var menuItemRightEdge = menuItem.offsetLeft + menuItem.clientWidth;
+      var maxVisiblePoint = this.tabsNavigationElement.scrollLeft + this.tabsNavigationElement.clientWidth;
+      // Element overflows from the currently visible navigation area. Adjust the
+      // navigation's scroll value so that the whole menu item fits inside the visible area.
+      if (menuItemRightEdge > maxVisiblePoint) {
+        this.tabsNavigationElement.scrollLeft += menuItemRightEdge - maxVisiblePoint;
```

Comment:
How come this is += rather than -=? Trying to wrap my head around this.

Reply:
That case is handled in the condition above this one. This part handles the cases in which the element is to the right of the visible area, so we have to move the scroll position right so that the element moves left.